File: dotnet/src/HybridRow.Tests.Unit/CustomerSchema/Address.cs
// ------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All rights reserved.
// ------------------------------------------------------------

#pragma warning disable SA1401 // Fields should be private

namespace Microsoft.Azure.Cosmos.Serialization.HybridRow.Tests.Unit.CustomerSchema
{
    internal sealed class Address
    {
        public string Street;
        public string City;
        public string State;
        public PostalCode PostalCode;

        public override bool Equals(object obj)
        {
            if (object.ReferenceEquals(null, obj))
            {
                return false;
            }

            if (object.ReferenceEquals(this, obj))
            {
                return true;
            }

            return obj is Address && this.Equals((Address)obj);
        }

        public override int GetHashCode()
        {
            unchecked
            {
                int hashCode = this.Street != null ? this.Street.GetHashCode() : 0;
                hashCode = (hashCode * 397) ^ (this.City != null ? this.City.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (this.State != null ? this.State.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (this.PostalCode != null ? this.PostalCode.GetHashCode() : 0);
                return hashCode;
            }
        }

        private bool Equals(Address other)
        {
            return string.Equals(this.Street, other.Street) &&
                   string.Equals(this.City, other.City) &&
                   string.Equals(this.State, other.State) &&
                   object.Equals(this.PostalCode, other.PostalCode);
        }
    }
}

File: jre/src/test/java/azure/data/cosmos/serialization/hybridrow/unit/customerschema/Address.java
//------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All rights reserved.
//------------------------------------------------------------

package azure.data.cosmos.serialization.hybridrow.unit.customerschema;

import azure.data.cosmos.serialization.hybridrow.Tests.Unit.*;

// TODO: C# TO JAVA CONVERTER: There is no preprocessor in Java:
///#pragma warning disable SA1401 // Fields should be private


public final class Address {
    public String City;
    public PostalCode PostalCode;
    public String State;
    public String Street;

    @Override
    public boolean equals(Object obj) {
        if (null == obj) {
            return false;
        }

        if (this == obj) {
            return true;
        }

        return obj instanceof Address && this.equals((Address)obj);
    }

    @Override
    public int hashCode() {
        // TODO: C# TO JAVA CONVERTER: There is no equivalent to an 'unchecked' block in Java:
        unchecked
        {
            int hashCode = this.Street != null ? this.Street.hashCode() : 0;
            hashCode = (hashCode * 397) ^ (this.City != null ? this.City.hashCode() : 0);
            hashCode = (hashCode * 397) ^ (this.State != null ? this.State.hashCode() : 0);
            hashCode = (hashCode * 397) ^ (this.PostalCode != null ? this.PostalCode.hashCode() : 0);
            return hashCode;
        }
    }

    private boolean equals(Address other) {
        return this.Street.equals(other.Street) && this.City.equals(other.City) && this.State.equals(other.State) && this.PostalCode.equals(other.PostalCode);
    }
}